slack/web/classes/elements.py

```diff
@@ -1,478 +1,1042 @@
         )
+        show_unknown_key_warning(__name__, others)
+
+        self.options = Option.parse_all(options)
+        self.option_groups = OptionGroup.parse_all(option_groups)
+        self.initial_options = Option.parse_all(initial_options)
+        self.max_selected_items = max_selected_items
+
+    @JsonValidator(f"options attribute cannot exceed {options_max_length} elements")
+    def validate_options_length(self):
+        return self.options is None or len(self.options) <= self.options_max_length
+
+    @JsonValidator(
+        f"option_groups attribute cannot exceed {option_groups_max_length} elements"
+    )
+    def validate_option_groups_length(self):
+        return (
+            self.option_groups is None
+            or len(self.option_groups) <= self.option_groups_max_length
+        )
+
+    @JsonValidator(f"options and option_groups cannot both be specified")
+    def validate_options_and_option_groups_both_specified(self):
+        return self.options is None or self.option_groups is None
+
+    @JsonValidator(f"options or option_groups must be specified")
+    def validate_neither_options_or_option_groups_is_specified(self):
+        return self.options is not None or self.option_groups is not None
+
+
+# SelectElement will be deprecated in version 3, use StaticSelectElement instead
+class SelectElement(InputInteractiveElement):
+    type = "static_select"
+    options_max_length = 100
+    option_groups_max_length = 100
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union({"options", "option_groups", "initial_option"})
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        placeholder: Optional[str] = None,
+        options: Optional[List[Option]] = None,
+        option_groups: Optional[List[OptionGroup]] = None,
+        initial_option: Optional[Option] = None,
+        confirm: Optional[Union[dict, ConfirmObject]] = None,
+        **others: dict,
+    ):
+        """This is the simplest form of select menu, with a static list of options passed in when defining the element.
+        https://api.slack.com/reference/block-kit/block-elements#static_select
+        """
+        super().__init__(
+            type=self.type,
+            action_id=action_id,
+            placeholder=TextObject.parse(placeholder, PlainTextObject.type),
+            confirm=ConfirmObject.parse(confirm),
+        )
+        show_unknown_key_warning(__name__, others)
+
         self.options = options
+        self.option_groups = option_groups
         self.initial_option = initial_option
 
     @JsonValidator(f"options attribute cannot exceed {options_max_length} elements")
-    def options_length(self):
-        return len(self.options) <= self.options_max_length
+    def validate_options_length(self):
+        return self.options is None or len(self.options) <= self.options_max_length
 
-    def to_dict(self) -> dict:
```

Comment:
Overriding this method is no longer necessary. If `attributes` is correctly set, the `to_dict` method works properly with any types of fields.